react-components/src/components/RevealToolbar/LayersContainer/PointCloudLayersContainer.tsx

```diff
@@ -79,19 +72,24 @@ export const PointCloudLayersContainer = (): ReactElement => {
   };
 
   return (
-    <Menu.Submenu content={pointCloudModelContent()} title="Point clouds">
-      <Flex direction="row" justifyContent="space-between">
-        <Checkbox
-          checked={allPointCloudModelVisible}
-          indeterminate={indeterminate}
-          onChange={(e, c) => {
-            e.stopPropagation();
-            handleAllPointCloudModelsVisibility(c as boolean);
-          }}
-        />
-        <StyledLabel> Point clouds </StyledLabel>
-        <StyledChipCount label={count} hideTooltip type="neutral" />
-      </Flex>
-    </Menu.Submenu>
+    <>
+      {pointCloudModels.length > 0 && (
+        <Menu.Submenu openOnHover={false} content={pointCloudModelContent()} title="Point clouds">
+          <Flex direction="row" justifyContent="space-between">
+            <Checkbox
+              key={`allPointCLoudModelCheckbox-${String(allModelVisible)}-${String(indeterminate)}`}
+              checked={allModelVisible}
+              indeterminate={indeterminate}
+              onChange={(e, c) => {
+                e.stopPropagation();
+                handleAllPointCloudModelsVisibility(c as boolean);
```

Comment:
Why is `c` not typed here?

Reply:
`c` in `onChange` is `string | boolean | undefined`

Reply:
Ok, so what happens if this is actually a string then?